crates/starknet/src/starknet/mod.rs

```diff
@@ -159,6 +175,95 @@ impl Starknet {
 
         this.restart_pending_block()?;
 
+        // Re-execute transactions
+        if transactions.is_some() {
+            // is_some() is need here?
+            for (_hash, transaction) in transactions.unwrap_or_default().iter() {
+                match transaction.inner.clone() {
+                    Transaction::Declare(DeclareTransaction::Version0(_tx)) => {
```

Comment:
```suggestion
                    Transaction::Declare(DeclareTransaction::Version0(_)) => {
```

Reply:
fixed in aca53690763aa8ccba3a03b49830d8cc260ae093